interop/grpclb_fallback/client.go

```diff
@@ -0,0 +1,224 @@
+// +build !appengine
+// +build go1.11
+
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc"
+	_ "google.golang.org/grpc/balancer/grpclb"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/alts"
+	"google.golang.org/grpc/credentials/google"
+	testpb "google.golang.org/grpc/interop/grpc_testing"
+)
+
+var (
+	customCredentialsType         = flag.String("custom_credentials_type", "", "Client creds to use")
+	serverURI                     = flag.String("server_uri", "dns:///staging-grpc-directpath-fallback-test.googleapis.com:443", "The server host name")
+	unrouteLBAndBackendAddrsCmd   = flag.String("unroute_lb_and_backend_addrs_cmd", "", "Command to make LB and backend address unroutable")
+	blackholeLBAndBackendAddrsCmd = flag.String("blackhole_lb_and_backend_addrs_cmd", "", "Command to make LB and backend addresses blackholed")
+	testCase                      = flag.String("test_case", "",
+		`Configure different test cases. Valid options are:
+        fast_fallback_before_startup : LB/backend connections fail fast before RPC's have been made;
+        fast_fallback_after_startup : LB/backend connections fail fast after RPC's have been made;
+        slow_fallback_before_startup : LB/backend connections black hole before RPC's have been made;
+        slow_fallback_after_startup : LB/backend connections black hole after RPC's have been made;`)
+	infoLog  = log.New(os.Stderr, "INFO: ", log.Ldate|log.Ltime|log.Lshortfile)
+	errorLog = log.New(os.Stderr, "ERROR: ", log.Ldate|log.Ltime|log.Lshortfile)
+)
+
+func doRPCAndGetPath(client testpb.TestServiceClient, deadline time.Time) testpb.GrpclbRouteType {
```

Comment:
Would it make more sense to pass `timeout time.Duration` instead, and avoid all the `time.Now().Add()`s at the call sites?

Reply:
good point, I think so, done